### ISSUE TITLE
`setOptions` on `TwigField` will remove the template

### ISSUE THREAD

**Issue by mamazu** [Sylius/SyliusGridBundle#267]
## The problem

The `TwigField` is basically just a normal field with a default template. The problem is if you want to change the options of this field it will either have to specify the template again or you will lose it.

## Example

```php
<?php

$field = TwigField::create('some_field', '@Sylius/some_template.html.twig')
    ->setOptions(['vars' => ['labels' => 'path/to/label'])
;
```

Will result in the options being:
```
[
    'vars' => ['labels' => 'path/to/label']
]
```

and not:

```
[
    'template' => '@Sylius/some_template.html.twig',
    'vars' => ['labels' => 'path/to/label']
]
```

## Possible solution

The quick and easy answer would be to have the `TwigField` extend the `Field` and change the behavior of the `setOptions` function to merge with the template. This would be a big change though.

**Comment by loic425** [comment]
@mamazu You're right. FYI, there is a "addOption". @lchrusciel merged too quickly the notice about this issue in the doc.
But I'm in favour of the merge options with the setOptions, that was my initial idea when implementing the grid builder. I think it'd be better for DX as we already have users who have that issue.

**Comment by mamazu** [comment]
So extending it would be the best idea? I can make a merge request for that if you want.

**Comment by mamazu** [comment]
I just had a look at this ticket again. I have two ideas for this. Either we deprecate the `setOptions` and forward people to the `addOptions` call which already does the merging or we keep the set options (as this can be quite helpful if you really want to remove / change a lot of options at once). However we then should also add a warning if the current options contain a twig field which would be unset by setting the options again. Then the user can decide if that's the right course of action or not.

**Comment by loic425** [comment]
@mamazu On the Symfony OptionsResolver, the setDefaults merge current defaults with new ones.
I think it's more intuitive.
https://github.com/symfony/symfony/blob/7.4/src/Symfony/Component/OptionsResolver/OptionsResolver.php#L277

But in this case, maybe we should deprecate addOptions instead.

**Comment by mamazu** [comment]
Okay I'll do that.
